src/model_theory/bundled.lean

```diff
@@ -37,5 +37,43 @@ instance equiv_setoid : setoid (category_theory.bundled L.Structure) :=
   iseqv := ⟨λ M, ⟨equiv.refl L M⟩, λ M N, nonempty.map equiv.symm,
     λ M N P, nonempty.map2 (λ MN NP, NP.comp MN)⟩ }
 
+variable (T : L.Theory)
+
+namespace Theory
+
+/-- The type of nonempty models of a first-order theory. -/
+structure Model :=
+(carrier : Type w)
+[struc : L.Structure carrier]
+[is_model : T.model carrier]
+[nonempty' : nonempty carrier]
+
+attribute [instance] Model.struc Model.is_model Model.nonempty'
+
+namespace Model
+
+instance : has_coe_to_sort (T.Model) (Type w) := ⟨Model.carrier⟩
+
+/-- The object in the category of R-algebras associated to a type equipped with the appropriate
+typeclasses. -/
+def of (M : Type w) [L.Structure M] [M ⊨ T] [nonempty M] :
```

Comment:
I think you should add a second constructor with `M ⊨ T` explicit. I expect that will happen frequently.